sys/parsoid_proxy.js

```diff
@@ -0,0 +1,174 @@
+"use strict";
+
+const P = require('bluebird');
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+
+const mwUtil = require('../lib/mwUtil');
+
+const spec = HyperSwitch.utils.loadSpec(`${__dirname}/parsoid.yaml`);
+
+class ParsoidProxy {
+    constructor(options) {
+        this.options = options || {};
+        if (this.options.backends) {
```

Comment:
`options.parsoid.backends` is passed in here, so this will never be true. You need `this.options`

Reply:
Fixed.